api_shield.go

```diff
@@ -0,0 +1,65 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+// AuthIdCharacteristics a single option from
+// configuration?properties=auth_id_characteristics.
+type AuthIdCharacteristics struct {
+	Type string `json:"type"`
+	Name string `json:"name"`
+}
+
+// APIShield is all the available options under
+// configuration?properties=auth_id_characteristics.
+type APIShield struct {
+	AuthIdCharacteristics []AuthIdCharacteristics `json:"auth_id_characteristics"`
+}
+
+// APIShieldResponse represents the response from the api_gateway/configuration endpoint.
+type APIShieldResponse struct {
+	Result APIShield `json:"result"`
+	Response
+	ResultInfo `json:"result_info"`
+}
+
+func (api *API) GetAPIShieldConfiguration(ctx context.Context, rc *ResourceContainer) (APIShield, ResultInfo, error) {
+	uri := fmt.Sprintf("/zones/%s/api_gateway/configuration?properties=auth_id_characteristics", rc.Identifier)
+
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
```

Comment:
do you know if there are other options here? i'm leaning towards making this it's own set of parameters to ensure that we can pivot the API with breaking changes but :shrug:

Reply:
AFICT its only the `auth_id_characteristics` https://api.cloudflare.com/#api-shield-settings-properties